cmd/sippy/loadocpvariants.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/openshift/sippy/pkg/dataloader/prowloader/gcs"
+	"github.com/openshift/sippy/pkg/variantregistry"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"google.golang.org/api/option"
+
+	"github.com/openshift/sippy/pkg/flags"
+)
+
+type LoadJobVariantsFlags struct {
+	BigQueryFlags    *flags.BigQueryFlags
+	GoogleCloudFlags *flags.GoogleCloudFlags
+	OutputFile       string
+}
+
+func NewLoadJobVariantsFlags() *LoadJobVariantsFlags {
+	return &LoadJobVariantsFlags{
+		BigQueryFlags:    flags.NewBigQueryFlags(),
+		GoogleCloudFlags: flags.NewGoogleCloudFlags(),
+	}
+}
+
+func (f *LoadJobVariantsFlags) BindFlags(fs *pflag.FlagSet) {
+	f.BigQueryFlags.BindFlags(fs)
+	f.GoogleCloudFlags.BindFlags(fs)
+	fs.StringVar(&f.OutputFile, "o", "ocp-expected-job-variants.json", "Output json file for job variant data")
+}
+
+func NewLoadOCPJobVariantsCommand() *cobra.Command {
+	f := NewLoadJobVariantsFlags()
+
+	cmd := &cobra.Command{
+		Use:   "load-ocp-job-variants",
```

Comment:
Does this command have to be OCP specific? Could it be `load-job-variants` and take a mode flag? Similar to how the variant registry works https://github.com/openshift/sippy/blob/master/pkg/testidentification/variants.go#L9

Future us would probably appreciate leaving the design more open if we productize this part

Reply:
My intention was to flag it clearly as ocp specific so third parties get a clear signal this is not for them. The intention was that the hook point is the file this generates, a non-ocp sippy user can do whatever they like to generate that file, and then the second command is reusable to get it all synced to bigquery on any cadence they wish.

QE might have very different rules, we probably would take that code in this repo, so perhaps --mode is a better idea. I'll get it switched.